ansible_rulebook/util.py

```diff
@@ -380,3 +380,32 @@ def validate_url(url: str, url_type: str) -> bool:
     if url_type == "controller":
         return res.scheme in ["http", "https"] and bool(res.netloc)
     return res.scheme in ["ws", "wss"] and bool(res.netloc)
+
+
+KEYS_TO_FILTER = [
+    "CONTROLLER_PASSWORD",
+    "TOWER_PASSWORD",
+    "AAP_PASSWORD",
+    "CONTROLLER_OAUTH_TOKEN",
+    "TOWER_OAUTH_TOKEN",
+    "AAP_OAUTH_TOKEN",
+    "postgres_db_password",
+]
+MASKED_VARIABLE = "******"
+
+
+def mask_sensitive_variable(key: str, val: str):
+    if any(
+        filtered_key.casefold() == str(key).casefold()
+        for filtered_key in KEYS_TO_FILTER
+    ):
+        return MASKED_VARIABLE
+    else:
+        return val
+
+
+def mask_sensitive_variable_values(variables):
+    return {
+        key: mask_sensitive_variable(key, val)
+        for key, val in variables.items()
```

Comment:
@zkayyali812 What happens with nested dictionaries?

Reply:
Updated to consider this case